test/catalog_test.exs

```diff
@@ -34,6 +34,21 @@ defmodule Wanda.CatalogTest do
       end)
     end
 
+    test "should return the whole catalog but the check with the when clause" do
+      catalog_path = Application.fetch_env!(:wanda, Wanda.Catalog)[:catalog_path]
+
+      valid_files =
```

Comment:
I think we could remove this `valid_files` code together with the `assert length(valid_files) - 1 == length(catalog)` line, this is already tested.
We should aim to see that the when clauses that don't match are not returned, which is already done with the latest assert. I think it should be enough